webapp/src/pages/Settings.tsx

```diff
@@ -2,58 +2,618 @@ import {
+                  "name"
+                ),
+              })
+            : setPartialConfig({
+                ...partialConfig,
+                pipelines: _.unionBy(
+                  updatedPipeline,
+                  resultingConfig?.pipelines,
+                  "name"
+                ),
+              });
+        }}
+      />
+      <Typography textTransform="capitalize" variant="caption" fontSize={15}>
+        Postprocessors
+      </Typography>
+    </Box>
+  );
+
+  const switchNullOrDefault = (
+    field: keyof AzimuthConfig,
+    selected: boolean
+  ) => (
     <Checkbox
-      checked={Boolean(resultingConfig[field])}
+      sx={{ paddingLeft: 0 }}
+      checked={selected}
       disabled={isError || isFetching}
       onChange={(_, checked) =>
         setPartialConfig({ ...partialConfig, [field]: checked ? {} : null })
       }
     />
   );
 
+  const displayReadonlyFields = (label: string, value: string | undefined) => (
+    <TextField
+      key={String(value)}
+      sx={{
+        m: 1,
+        width: "32ch",
+      }}
+      size="small"
+      variant="standard"
+      label={<Typography fontWeight="bold">{label}</Typography>}
+      value={value}
+      disabled={isError || isFetching}
+      InputProps={{
+        readOnly: true,
+        disableUnderline: true,
+        style: { textTransform: "capitalize", fontSize: 14 },
+      }}
+    />
+  );
+
+  const displaySubFields = (
+    config: keyof AzimuthConfig,
+    field: string,
+    value: number | undefined
+  ) => (
+    <TextField
+      key={field}
+      sx={{ m: 1, width: "16ch" }}
+      size="small"
+      label={field}
+      type="number"
+      defaultValue={value}
+      disabled={!Boolean(resultingConfig[config])}
+      inputProps={STEPPER[field]}
+      InputProps={{ style: { fontSize: 14 } }}
+      variant="standard"
+      onChange={(event) =>
+        setPartialConfig(
+          _.merge(partialConfig, {
+            [config]: { [field]: Number(event.target.value) },
+          })
+        )
+      }
+    />
+  );
+
+  const displayPostprocessorSubField = (
+    pipeline: PipelineDefinition,
+    field: string,
+    postprocessorIdx: number,
+    value: number | undefined
+  ) => (
+    <TextField
+      key={field}
+      sx={{ m: 1, width: "16ch" }}
+      size="small"
+      label={field}
+      type="number"
+      InputLabelProps={{
+        shrink: true,
+      }}
+      defaultValue={value}
+      inputProps={{ min: 0, max: 1, step: 0.1 }}
+      variant="standard"
+      onChange={(event) => {
+        const existedPipeline = partialConfig.pipelines
+          ? partialConfig.pipelines.find(({ name }) => name === pipeline.name)
+          : undefined;
+        const current_postprocessor = pipeline?.postprocessors?.find(
+          (_, index) => index === postprocessorIdx
+        );
+        const updatedPostprocessor = _.assign({}, current_postprocessor, {
+          [field]: Number(event.target.value),
+        });
```

Comment:
This use of `_.assign` only results in a shallow copy (not deep), which might be fine, so it is equivalent to
```suggestion
        const updatedPostprocessor = {
          ...currentPostprocessor,
          [field]: Number(event.target.value),
        };
```